Cache.js

```diff
@@ -152,9 +152,12 @@ define([
 				// the result from the put should be dictated by the master store and be unaffected by the cachingStore,
 				// unless the master store doesn't implement put
 				return result || cachedPutResult;
-			}, function () {
+			}, function (exception) {
 				// Assuming a rejection of a promise invalidates the local cache
-				cachingStore.remove((directives && directives.id) || this.getIdentity(object));
+				cachingStore.remove((directives && directives.id) || cachingStore.getIdentity(object));
+				if(exception) {
```

Comment:
This is propagating the rejection only when the promise was rejected with a value or object.  Wouldn't we want it to always throw an error?

Reply:
agree. I added an additional 'throw'